pipenv/routines/update.py

```diff
@@ -114,13 +186,33 @@ def upgrade(
     lock_only=False,
     extra_pip_args=None,
 ):
+    """Enhanced upgrade command with dependency conflict detection."""
     lockfile = project.lockfile()
     if not pre:
         pre = project.settings.get("allow_prereleases")
-    if dev or "dev-packages" in categories:
-        categories = ["develop"]
-    elif not categories or "packages" in categories:
-        categories = ["default"]
+    if not categories:
```

Comment:
If the original authors had just made the package group names identical between Pipfile and lock it would have saved 50% of my headache here.